Team9Project/src/com/indragie/cmput301as1/ExpenseClaimListActivity.java

```diff
@@ -200,21 +239,74 @@ private void onManageTagsResult(Intent data) {
 				switch (mutation.getMutationType()) {
 				case DELETE:
 					claim.removeTag(tag);
+					if(filteredTagsList.contains(tag) && checkFilteredTags()) {
+						filteredTagsList.remove(tag);
+					}
 					break;
 				case EDIT:
 					int tagIndex = claim.getTags().indexOf(tag);
 					claim.setTag(tagIndex, mutation.getNewTag());
+					if(filteredTagsList.contains(tag) && checkFilteredTags()) {
+						int selectedIndex = filteredTagsList.indexOf(tag);
+						filteredTagsList.set(selectedIndex, mutation.getNewTag());
+					}
 					break;
 				}
 				
 				modifiedClaims.append(i, claim);
 				i++;
 			}
+			
 		}
 		
 		for (int i = 0; i < modifiedClaims.size(); i++) {
 			listModel.set(modifiedClaims.keyAt(i), modifiedClaims.valueAt(i));
 		}
+		if(checkFilteredTags()) {
+			setFilteredClaims();
+		}
+		
+	}
+	
+	/**
+	 * Sets the listModel used to filteredListModel.
+	 * @param data The intent to get the filteredTagsList.
+	 */
+	@SuppressWarnings("unchecked")
+	private void onFilterTagsRequest(Intent data) {
+		//TODO: Filter tags, replace the tags, etc.
+		filteredTagsList = (ArrayList<Tag>)data.getSerializableExtra(FilterTagsActivity.TAG_TO_FILTER);
+
+		setFilteredClaims();
```

Comment:
Couldn't you just pass `filteredTagsList` into `setFilteredClaims()` as a parameter rather than having to make it a property on the class?